### ISSUE TITLE
FCL cannot find libccd installed on $HOME/.local

### ISSUE THREAD

**Issue by xinyazhang** [flexible-collision-library/fcl#497]
I installed my custom built libccd under `$HOME/.local/lib/x86_64-linux-gnu/` (because I don't have the root access to the machine I'm using), and current FCL 0.6.1 cannot find the installed libccd even if `pkg_check_modules(PC_CCD ccd)` sets `PC_CCD_FOUND` be 1.

After digging into this problem for a while I found
1. `pkg-config -cflags ccd` would [print empty if the include directory is one of the default locations.](https://unix.stackexchange.com/questions/362289/why-is-pkg-config-cflags-openssl-returning-nothing-on-rhel-6-8)
2. `pkg_check_moudules` uses `pkg-config -cflags` to locate the include dir
3. The FCL's `CMakeLists.txt` uses `<prefix>_INCLUDE_DIRS` to find the header, which depends on the output of `pkg-config -cflags`
4. Thus, if libccd is installed at one of the default locations and cannot be found through `find_package`, the `pkg_config` method will fail because `<prefix>_INCLUDE_DIRS` is empty.

**Comment by v4hn** [comment]
I'm just another user of FCL, but also a linux distribution maintainer overlaying modules every day. Your problem is not with FCL, but with your own installation.

FCL expects to find a `ccd-config.cmake` or a `ccd.pc` file that point to the version of libccd that you want to use. You can't just copy an `so` file and/or headers to a folder and expect cmake to find them.
Instead it is your responsibility to make sure these files correctly overlay their respective system-wide versions (if those exist).

To make pkg-config aware of files in different locations, you can add the folder to the `PKG_CONFIG_PATH` environment variable, e.g., `export PKG_CONFIG_PATH="$INSTALL_PREFIX/lib/pkgconfig:$PKG_CONFIG_PATH"`.

For cmake it's somewhat counterintuitive, but you have to extend `PATH` with **an existing** `$INSTALL_PREFIX/bin` folder.
cmake will automagically look for `$INSTALL_PREFIX/lib.*/ccd/ccd-config.cmake` (and all other valid locations) to find the config when `find_package(ccd)` is invoked.

Of course the files have to provide the correct installations locations as well, but they usually do if you build with `-DCMAKE_INSTALL_PREFIX=<your location here>`.

**Comment by xinyazhang** [comment]
> To make pkg-config aware of files in different locations, you can add the folder to the `PKG_CONFIG_PATH` environment variable, e.g., `export PKG_CONFIG_PATH="$INSTALL_PREFIX/lib/pkgconfig:$PKG_CONFIG_PATH"`.

No this is not the case.
I would emphasis again, in my configuration the `pkg-config` are working totally fine, and my PKG_CONFIG_PATH is correctly set.
![image](https://user-images.githubusercontent.com/11429462/95411995-8e251b80-08ed-11eb-81e9-c2e869d30603.png)
I added `message` to `CMakeLists.txt` for debugging, and after running cmake I found.
![image](https://user-images.githubusercontent.com/11429462/95412536-c8db8380-08ee-11eb-85e8-3f2d6169489d.png)

In other words, `<prefix>_INCLUDE_DIRS` is not reliable because it could be empty on some system, and `CMakeLists.txt` should use `<prefix>_FOUND` to determine the existence of library.

**Comment by xinyazhang** [comment]
Further details: this behaviour of `pkg-config` is implemented at [LN 802 -- 865 in pkg.c](https://cgit.freedesktop.org/pkg-config/tree/pkg.c?h=pkg-config-0.29.1&id=87152c05be88ca8be71a3a563f275b3686d32c28#n854), where `pkg-config` removes any `-I` cflags if it's already in `$PKG_CONFIG_SYSTEM_INCLUDE_PATH`, `$C_INCLUDE_PATH`, or `$CPLUS_INCLUDE_PATH`

**Comment by v4hn** [comment]
> In other words, <prefix>_INCLUDE_DIRS is not reliable because it could be empty on some system, and CMakeLists.txt should use <prefix>_FOUND to determine the existence of library.

You are right. Your initial description did not link to [the respective code lines](https://github.com/flexible-collision-library/fcl/blob/master/CMakeLists.txt#L194) and checking the respective `*CCD_FOUND` variables is the correct way to check for successful detection.

This whole block seems to be a a hack [for multiple package config names](https://github.com/flexible-collision-library/fcl/blob/master/CMakeLists.txt#L177-L178) and could be avoided by just checking for the different package names sequentially.

Could you prepare a pull-request to fix it?

**Comment by xinyazhang** [comment]
Sure. I actually already have a dirty hack over this problem. A cleaner solution  is on the way.
